File: src/main/resources/public/js/app/service/userService.js
app.service('userService', function ($http) {
    this.list = function (text, size, page) {
        return $http.get('api/users', {
            params: {
                text: text || '',
                page: page || 0,
                size: size || 10
            }
        })
    };

    this.delete = function (id) {
        return $http.delete('api/users/' + id);
    };

    this.currentUser = function () {
        return $http.get('api/users/currentUser');
    };
    this.save = function (user) {
        return $http.post('auth/register', user);
    }
});

File: src/main/java/app/users/UserService.java
package app.users;


import app.core.BaseService;
import app.security.CustomUserDetails;

import java.security.Principal;

public interface UserService extends BaseService<User,Long> {
    Object findAll(Long size, Long page, String text);
    User findByLogin(String value);
    User save(User user, CustomUserDetails userDetails);
    User currentUser(Principal principal);
}
